pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -39,7 +39,7 @@ func init() {
 	// Register functions that extract metadata used by predicates and priorities computations.
 	factory.RegisterPredicateMetadataProducerFactory(
 		func(args factory.PluginFactoryArgs) algorithm.PredicateMetadataProducer {
-			return predicates.NewPredicateMetadataFactory()
+			return predicates.NewPredicateMetadataFactory(args.PodLister)
```

Comment:
Is that argument even necessary? @bsalamat can this be removed?

Reply:
I think this can be removed. If yes, I can do that in a separate PR in 1.11.